Isosurfacing_3/examples/Isosurfacing_3/dual_contouring_mesh_offset.cpp

```diff
@@ -0,0 +1,85 @@
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/Surface_mesh.h>
+
+#include <CGAL/Isosurfacing_3/dual_contouring_3.h>
+#include <CGAL/Isosurfacing_3/Implicit_Cartesian_grid_domain_3.h>
+
+#include <CGAL/AABB_face_graph_triangle_primitive.h>
+#include <CGAL/AABB_traits.h>
+#include <CGAL/AABB_tree.h>
+#include <CGAL/Bbox_3.h>
+#include <CGAL/Side_of_triangle_mesh.h>
+
+#include <CGAL/boost/graph/IO/OFF.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+using Kernel = CGAL::Simple_cartesian<double>;
+using FT = typename Kernel::FT;
+using Point = typename Kernel::Point_3;
+using Vector = typename Kernel::Vector_3;
+
+using Mesh = CGAL::Surface_mesh<Point>;
+
+using Primitive = CGAL::AABB_face_graph_triangle_primitive<Mesh>;
+using Traits = CGAL::AABB_traits<Kernel, Primitive>;
+using Tree = CGAL::AABB_tree<Traits>;
+
+using Point_range = std::vector<Point>;
+using Polygon_range = std::vector<std::vector<std::size_t> >;
+
+int main(int, char**)
+{
+  const std::string input_name = CGAL::data_file_path("meshes/cross.off");
+  const Vector grid_spacing(0.1, 0.1, 0.1);
```

Comment:
Just playing around I changed the grid spacing to `(0.1, 0.02, 0.01)`, which results in this 
![image](https://github.com/CGAL/cgal/assets/3263539/f4ab758e-b455-4af3-bb35-c204c311b650)

Do I choose parameters which are not coherent with other parameters,  is that the expected result, or is it an indication for a potential bug?

Reply:
I tried debugging it for a few hours yesterday, I don't think it's a bug in the implementation.
I'm not sure what the issue is however, first I thought it was a numerical instability, so I implemented a slower but more stable approach. That didn't help however, so I thought maybe the mathematical formulation only works for cube cells, but that also shouldn't be the case. The issue only appears if the cells have high aspect ratios, it seems to work up to about 1:0.6